build_resolvers/lib/src/analysis_driver.dart

```diff
@@ -31,6 +34,11 @@ Future<AnalysisDriverForPackageBuild> analysisDriver(
     ),
     resourceProvider: buildAssetUriResolver.resourceProvider,
     sdkSummaryBytes: File(sdkSummaryPath).readAsBytesSync(),
+    byteStore: createByteStore(
+      directory: cachePath,
+      maxFileSize: 1024 * 1024 * 1024, // 1 GiB
```

Comment:
this could likely be increased some, although I don't have a sense of how large this cache would get on big projects

Reply:
The analysis server uses a 1GiB cache as well. That is for pretty much everything the analyzer has ever seen, so I think the limitation is already fairly large here.

Since we previously used a complete in-memory byte store, I have now chosen an approach that doesn't have a limitation on how much we keep in memory as well. We also write everything into a single file in the end instead of doing a write for every cache change. The analyzer also spawns an isolate doing regular cleanup work for each byte store, so I think a first-party implementation in `build_resolvers` that knows when to persist data is better.